src/main/java/codechicken/nei/ClientHandler.java

```diff
@@ -165,16 +154,19 @@ public static void loadSerialHandlers() {
         if (!file.exists()) {
             try (FileWriter writer = new FileWriter(file)) {
                 NEIClientConfig.logger.info("Creating default serial handlers list {}", file);
-                Collection<String> toSave = Loader.isModLoaded("dreamcraft") ? Collections.singletonList("")
-                        : Arrays.asList(defaultSerialHandlers);
-                IOUtils.writeLines(toSave, "\n", writer);
+                URL defaultSerialHandlersResource = ClientHandler.class
+                        .getResource("/assets/nei/cfg/serialhandlers.cfg");
```

Comment:
(the following also applies to your comment below about `heighthackhandlers.cfg`)

Uh, I think you may be misunderstanding the code here? The idea was to not change the externally observable behavior of NEI. Before this change, NEI would check if a `serialhandlers.cfg` file already exists in the config dir (`config/NEI`), if it doesn't, NEI would initialize that file from an array of strings. After that, in a separate step, NEI reads and tries to parse the contents of the file in `config/NEI`.

The only thing that I have changed here is the initialization step: instead of initializing it from an array of strings, it's copied from a dedicated resource file. The resource file is not parsed in any way, the single source of truth is the file in the config dir, only the way this file is initialized has been changed.

The reason for the change is to make it easier to extend the file for someone not familiar with the java syntax. Being a dedicated file it's easier to find now as well. No need to trawl java code.

> Hmm, do we need to have a separate file here?

There is no separate file here. The only file that matters is `config/NEI/sericalhandlers.cfg`.

Reply:
Ah, sorry, you're right. I was thinking that we could just include the defaults in the config file itself, but that wouldn't work if we wanted to ship those defaults with NEI itself, so we do need this separate assets file.